### PR TITLE
Team feature project

### PR DESCRIPTION
**Pull request by ttseng**
https://team-feature-collection.glitch.me/

(sorry this has nothing to do with collections, but I'm worried that changing the name will mess up syncing with the project, so I'm leaving the project name for now)

Added featured project embed to team pages.

Note - also fixed a bug with filtering of pinned projects on team pages. (You can notice it here: https://glitch.com/@spotify)

Also renamed the "Remove Project" button to "Remove project from team" to clarify that the act is non-destructive.

### CODE REVIEW

File: src/presenters/pages/team.jsx [hunk 1]
```diff
@@ -103,6 +105,13 @@ class TeamPage extends React.Component {
   }
 
   render() {
+    
+    const pinnedSet = new Set(this.props.team.teamPins.map(({projectId}) => projectId));
+    const pinnedProjects = this.props.team.projects.filter( ({id}) => pinnedSet.has(id)).filter ( ({id}) => id != this.props.team.featuredProjectId);
```
Comment (clottman) [inline src/presenters/pages/team.jsx]: This section (109-112) is kind of hard to read, and I think there's probably some tiny performance optimizations we could make. 

What about storing the result of the union between pinnedSet and teamProjects so you don't have to repeat that check?

File: src/presenters/pages/team.jsx [hunk 2]
```diff
@@ -178,10 +187,22 @@ class TeamPage extends React.Component {
           />
         </ErrorBoundary>
         
+        {featuredProject && 
+          <>
```
Comment (clottman) [inline src/presenters/pages/team.jsx]: What's the fragment helping with here? I think should be able to get rid of it and directly do `{featuredProject && <EntityPageFeaturedProject />}`

File: src/presenters/team-editor.jsx [hunk 3]
```diff
@@ -206,6 +218,9 @@ class TeamEditor extends React.Component {
       updateUserPermissions: (id, accessLevel) => this.updateUserPermissions(id, accessLevel).catch(handleError),
       joinTeamProject: projectId => this.joinTeamProject(projectId).catch(handleError),
       leaveTeamProject: projectId => this.leaveTeamProject(projectId).catch(handleError),
+      addProjectToCollection: (project,collection) => this.addProjectToCollection(project, collection).catch(handleError),
```
Comment (clottman) [inline src/presenters/team-editor.jsx]: Where do these get used?

File: src/presenters/pages/team.jsx (hunk 1 above)
Comment (ttseng) [reply src/presenters/pages/team.jsx]: made some changes to store both pinned projects + the featured project in a set, and then filtering by the set.  do you have any other suggestion to make it more efficient?

File: src/presenters/pages/team.jsx (hunk 2 above)
Comment (ttseng) [reply src/presenters/pages/team.jsx]: yup, I removed it

File: src/presenters/pop-overs/add-team-project-pop.jsx [hunk 4]
```diff
@@ -88,10 +89,14 @@ export class AddTeamProjectPop extends React.Component {
     return await this.props.api.post(inviteUserPath);
   }
 
-  onClick(event, project) {
+  onClick(event, project, createNotification) {
     event.preventDefault();
     this.props.togglePopover();
     this.props.addProject(project);
+    
+    console.log('create notification');
```
[the remark this answers is not in the document]
Comment (ttseng) [reply src/presenters/pop-overs/add-team-project-pop.jsx]: yup, removed

File: src/presenters/pop-overs/add-team-project-pop.jsx [hunk 5]
```diff
@@ -88,10 +89,14 @@ export class AddTeamProjectPop extends React.Component {
...
+    
+    createNotification(<p>Added <b><span className="project-name">{project.domain}</span></b></p>, "notifySuccess");
```
[the remark this answers is not in the document]
Comment (ttseng) [reply src/presenters/pop-overs/add-team-project-pop.jsx]: I created a new function in Notification.jsx for this.

File: src/presenters/pages/team.jsx (hunk 1 above)
Comment (ttseng) [reply src/presenters/pages/team.jsx]: ok - made better with lodash partition: `const [pinnedProjects, recentProjects] = partition(user.projects, ({id}) => pinnedAndFeaturedSet.has(id));`

File: src/presenters/pages/team.jsx [hunk 6]
```diff
@@ -153,35 +162,45 @@ class TeamPage extends React.Component {
                   inviteEmail={this.props.inviteEmail}
                   inviteUser={this.props.inviteUser}
                   setWhitelistedDomain={this.props.currentUserIsTeamAdmin ? this.props.updateWhitelistedDomain : null}
-                  members={this.props.team.users.map(({id}) => id)}
-                  whitelistedDomain={this.props.team.whitelistedDomain}
+                  members={team.users.map(({id}) => id)}
```
Comment (clottman) [inline src/presenters/pages/team.jsx]: Just an fyi as I see you using this pattern a lot - you can just call `_.map(team.users, 'id')` if you'd like! No need to change it now though.

File: src/presenters/team-editor.jsx (hunk 3 above)
Comment (clottman) [reply src/presenters/team-editor.jsx]: I guess it's probably out of scope for this PR, but the setup here seems really odd. I'll flag it for @sheridanvk & others to look at, though.

File: webpack.config.js [hunk 7]
```diff
@@ -30,7 +30,7 @@ module.exports = {
     path: BUILD,
     publicPath: '/',
   },
-  devtool: mode === 'production' ? 'source-map' : 'eval',
+  devtool: mode === 'production' ? 'source-map' : 'cheap-module-source-map',
```
Comment (clottman) [inline webpack.config.js]: Can you confirm with @sheridanvk or @GregWeil that this is an intended change and not a regression caused by git being silly?

File: src/presenters/pages/team.jsx (hunk 6 above)
Comment (ttseng) [reply src/presenters/pages/team.jsx]: oh cool, that's definitely more concise. thanks!

File: webpack.config.js (hunk 7 above)
Comment (ttseng) [reply webpack.config.js]: this matches what's currently on production

**Pull request merged**